database.php

```diff
@@ -72,7 +71,37 @@ function __construct($config) {
         // create database file when not existent yet
         $this->db = new PDO($db_input, $user, $password);
         $this->db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
-        $this->db->exec("CREATE TABLE IF NOT EXISTS registrations(
+	if (strpos($db_input, 'mysql:') !== false) {
+          $this->db->exec("CREATE TABLE IF NOT EXISTS `registrations` (
+	  		`id` int(10) UNSIGNED NOT NULL AUTO_INCREMENT,
+			`state` tinyint(4) NOT NULL DEFAULT '0',
+			`first_name` varchar(64) DEFAULT NULL,
+			`last_name` varchar(64) DEFAULT NULL,
+			`username` varchar(64) NOT NULL,
+			`password` varchar(64) NOT NULL,
+			`note` varchar(255) DEFAULT NULL,
+			`email` varchar(255) NOT NULL,
+			`verify_token` varchar(64) NOT NULL,
+			`admin_token` varchar(64) NOT NULL,
+			`request_date` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
+			PRIMARY KEY (`id`),
+			UNIQUE KEY `username` (`username`)
+			) ENGINE=MyISAM DEFAULT CHARSET=utf8mb4");
```

Comment:
I would not define an ENGINE here - or default to InnoDB as it is transaction-safe. Same below

Reply:
I don't know if it is still true, but I learned some time that if you don't need transactions and/or foreign key support use MyISAM because it has the smaller disk-footprint, is faster in most cases (many more reads than writes) and has the better tools for optimizing. There I would prefer MyISAM if there is no need for transactions and/or foreign keys. If there is a need for it, there should InnoDB explicitly selected.